tests/5.0/target_teams_distribute/test_target_teams_distribute_reduction_and.c

```diff
@@ -0,0 +1,88 @@
+//===--- test_target_teams_distribute_reduction_and.c------------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// This test uses the reduction clause on a target teams distribute directive,
+// testing that the variable in the reduction clause is properly reduced using
+// the and operator.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+int test_and() {
+  char a[N];
+  char result;
+  char host_result;
+      // The below calculation is meant to ensure about half the test arrays will
+      // come out to true under the 'and' operator, and the rest false. For the and
+      // operator, a test array that comes out true requires every entry to be true, 
+      // which is why this margin is so close to 100%.
+  double false_margin = pow(exp(1), log(.5)/N);
+  int errors = 0;
+  int num_teams[N];
+  int warned = 0;
+  srand(1);
+
+  for (int itr_count = 0; itr_count < 16; ++itr_count) {
```

Comment:
I was wondering if we should change this for a while that checks that we will test for all true (and resulting in a `true` result. And for at least one false. 
Either way I am fine with this as well, it's just that there is a change we won't check for an all true result.

Reply:
I think this is a good idea, but should we be worried that the test might never terminate, since it relies on random behavior? Not sure.

Reply:
Yes, that's also a possibility, large threshold? `while (tested_all && num_attempts < THRESHOLD)`?